ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+    def check_required_fields(self, parser_file):
+        required_fields = ["ParserParams", "ParserQuery", "Normalization.Schema" ]
+        missing_fields = []
+        for full_field in required_fields:
+            file = parser_file
+            fields = full_field.split('.')
+            for field_name in fields:
+                if field_name not in file:
+                    missing_fields.append(full_field)
+                    break
+                file = file[field_name]
+        if len(missing_fields) != 0:
+            self.fail(f"The following fields are missing in the file:\n{missing_fields}")
+    
+
+    # Test for parameter which are not datetime,dynamic or disabled
+    def scalar_test(self, param, no_call_query, column_name_in_table):
+        param_name = param['Name']
+        no_filter_query = no_call_query + create_call_without_parameter(column_name_in_table)
+        no_filter_response = self.send_query(no_filter_query)
+        self.assertNotEqual(len(no_filter_response.tables[0].rows) , 0 , f"No data for parameter:{param_name}")
+        with  self.subTest():
+            self.assertNotEqual(len(no_filter_response.tables[0].rows), 1, f"Only one value exists for parameter: {param_name} - validations for this parameter are partial" )
+        # Taking the first value returned in the response
+        selected_value = no_filter_response.tables[0].rows[0][0]
+        value_to_filter = f"\'{selected_value}\'" if param['Type']=="string" else selected_value
+
+        # Performing a filtering by the first value returned in the first response
+        query_with_filter = no_call_query + create_call_with_parameter(param_name, value_to_filter, column_name_in_table)
+        if selected_value=="":
+            query_with_filter = no_call_query + f"query() | where isempty({column_name_in_table}) | summarize count() by {column_name_in_table}\n"
+        filtered_response = self.send_query(query_with_filter)
+        with self.subTest():
+            self.assertNotEqual(0, len(filtered_response.tables[0].rows), f"Parameter: {param_name} - Got no results at all after filtering. Filtered by value: {value_to_filter}")
+        with self.subTest():
+            self.assertEqual(1, len(filtered_response.tables[0].rows), f"Parameter: {param_name} - Expected to have results for only one value after filtering. Filtered by value: {value_to_filter}")
+
+        # Performing a query with a non-existing value, expecting to return no results
+        no_results_query = no_call_query + create_call_with_parameter(param_name, DUMMY_VALUE, column_name_in_table)
+        no_results_response = self.send_query(no_results_query)
+        with self.subTest():
+            self.assertEqual(0, len(no_results_response.tables[0].rows), f"Parameter: {param_name} - Returned results for non existing filter value. Filtered by value: {DUMMY_VALUE}")
+        
+
+        
+    def disabled_test(self, param, no_call_query):
```

Comment:
>disabled_test [](http://example.com/codeflow?start=8&length=13)

Add docs to this function and all other test functions

Reply:
Added docs to this function and to "scalar_test" function.